smoke-tests/apps/OtelResourceCustomMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OtelResourceCustomMetricTest.java

```diff
@@ -0,0 +1,116 @@
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8_OPENJ9;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.microsoft.applicationinsights.smoketest.schemav2.Data;
+import com.microsoft.applicationinsights.smoketest.schemav2.DataPoint;
+import com.microsoft.applicationinsights.smoketest.schemav2.Envelope;
+import com.microsoft.applicationinsights.smoketest.schemav2.MetricData;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@UseAgent
+abstract class OtelResourceCustomMetricTest {
+
+  private static final String OTEL_RESOURCE_ATTRIBUTES =
+      "cloud.resource_id=/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxdfcdaa1/resourceGroups/fake-aks-cluster-name/providers/Microsoft.ContainerService/managedClusters/aks-vanilla-1,cloud.region=eastus,k8s.cluster.name=aks-vanilla-1,k8s.pod.namespace=default,k8s.node.name=aks-agentpool-19737836-vmss000001,k8s.pod.name=customer-test-app-78d8bf887c-nplmk,k8s.pod.uid=efa8fdda-873c-4a6f-b02a-8d3b00bc34a7,k8s.container.name=test-app-java,cloud.provider=Azure,cloud.platform=azure_aks,k8s.replicaset.name=customer-java-very-big-756899c8b6,k8s.deployment.name=customer-java-very-big,k8s.replicaset.uid=69ef5d52-c770-4ec7-a3a2-2e2e8e885b3d";
+
+  @RegisterExtension
+  static final SmokeTestExtension testing =
+      SmokeTestExtension.builder().otelResourceAttributesEnvVar(OTEL_RESOURCE_ATTRIBUTES).build();
+
+  private static final int COUNT = 100;
+
+  @Test
+  @TargetUri(value = "/app", callCount = COUNT)
+  void testApp() {
+    List<Envelope> metricsEnvelops = testing.mockedIngestion.getItemsEnvelopeDataType("MetricData");
+    List<Envelope> otelResourceCustomMetrics = new ArrayList<>();
+    for (Envelope envelope : metricsEnvelops) {
+      MetricData metricData = (MetricData) ((Data<?>) envelope.getData()).getBaseData();
+      String name = metricData.getMetrics().get(0).getName();
+      if ("_OTELRESOURCE_".equals(name)) {
+        otelResourceCustomMetrics.add(envelope);
+      }
+    }
+
+    verify(otelResourceCustomMetrics);
+  }
+
+  private static void verify(List<Envelope> metrics) {
+    for (Envelope envelope : metrics) {
+      validateTags(envelope);
+      MetricData md1 = (MetricData) ((Data<?>) envelope.getData()).getBaseData();
+      validateMetricData(md1);
+    }
+  }
+
+  private static void validateTags(Envelope envelope) {
+    Map<String, String> tags = envelope.getTags();
+    assertThat(tags.get("ai.internal.sdkVersion")).isNotNull();
+    assertThat(tags).containsEntry("ai.cloud.roleInstance", "customer-test-app-78d8bf887c-nplmk");
+    assertThat(tags).containsEntry("ai.cloud.role", "customer-java-very-big");
```

Comment:
can you add another test to check that role name and role instance from application insights json take precedence over these?

Reply:
this is aks.. we have specific logic for aks. this is expected. that case has been covered in the rolename smoke test. it will take precedence over. i want to test aks logic here.

Reply:
which takes precedence?
* `k8s.deployment.name` resource attribute, or
* `APPLICATIONINSIGHTS_ROLE_NAME` env var

Reply:
APPLICATIONINSIGHTS_ROLE_NAME  or the json config.

Reply:
we map env var to service.name, which has the highest precedence over anything else. we have lots of tests about it already. this test is not about that.

Reply:
do we have a smoke test for this?

Reply:
otel endpoint smoke test